app/models.py

```diff
@@ -94,3 +95,62 @@ def facet_plural(facet: str):
         facet_plural = facet
 
     return facet_plural
+
+
+class ElementsItem(BaseModel):
+    element_type: str
+    text_element: Union[str, None] = None
+
+
+class KnowledgePanelItem(BaseModel):
+    title: str
+    subtitle: Union[str, None] = None
+    source_url: Union[str, None] = None
+    elements: Union[list[ElementsItem], None] = None
+
+
+class WikidataElementsItem(BaseModel):
+    element_type: str
+    image_url: Union[str, None] = None
+    wikipedia: Union[str, None] = None
+    open_street_map: Union[str, None] = None
+    INAO: Union[str, None] = None
+
+
+WikidataPanel = Union[ElementsItem, WikidataElementsItem]
+
+
+class WikidataKnowledgePanelItem(BaseModel):
+    title: str
+    subtitle: Union[str, None] = None
+    source_url: Union[str, None] = None
+    elements: Union[list[WikidataPanel], None] = None
+
```

Comment:
The current example schema for wikidata look like this 👇🏽 
![image](https://user-images.githubusercontent.com/96618001/188637286-87433fa1-9e78-42b1-953f-8b6a054309b7.png)
but when I run the it became like this 👇🏽 
![image](https://user-images.githubusercontent.com/96618001/188637440-dd2abae3-8580-4027-aa2c-e1a1b75d96f0.png)

Reply:
Ok, FTR the url to use is http://localhost/knowledge_panel?facet_tag=category&value_tag=en%3Abeers

The problem is that FastAPI do try to convert your dict to a Model, but it seems it does it in a naïve way and it's not able to tell the difference between the text element and the wikidata one.

But, for me, the correct way to do this is: in knowledge.py, instead of building dict, you should instanciate models and return that. So there is no magic, and also you are sure models won't miss any field from the dicts.